nnvm/python/nnvm/frontend/darknet.py

```diff
@@ -440,11 +440,13 @@ def __init__(self, net, dtype='float32'):
         self._state_ctr['cell_state'] = 0
         self._state_ctr['gru'] = 0
 
-    def _read_memory_buffer(self, shape, data):
+    def _read_memory_buffer(self, shape, data, dtype=None):
```

Comment:
initalize with self.dtype

Reply:
self will not be available at the time of initialization. One of the good way to handle such situation is initializing with None and checking in the method.